src/docker/engine.rs

```diff
@@ -0,0 +1,64 @@
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+use crate::errors::*;
+use crate::extensions::CommandExt;
+
+const DOCKER: &str = "docker";
+const PODMAN: &str = "podman";
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum EngineType {
+    Docker,
+    Podman,
+    PodmanRemote,
+    Other,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Engine {
```

Comment:
Add the engine with the engine type, so we can simplify this. Also adds `PodmanRemote`, which is a separate command and essentially encompasses the same functionality as `podman --remote`.